src/NServiceBus.RabbitMQ/Configuration/ConnectionConfiguration.cs

```diff
@@ -62,14 +149,51 @@ void SetDefaultClientProperties(string endpointName)
 
             var hostname = RuntimeEnvironment.MachineName;
 
-            ClientProperties.Add("client_api", "NServiceBus");
-            ClientProperties.Add("nservicebus_version", nsbFileVersion);
-            ClientProperties.Add("nservicebus.rabbitmq_version", rabbitMQFileVersion);
-            ClientProperties.Add("application", applicationName);
-            ClientProperties.Add("application_location", applicationPath);
-            ClientProperties.Add("machine_name", hostname);
-            ClientProperties.Add("user", UserName);
-            ClientProperties.Add("endpoint_name", endpointName);
+            var clientProperties = new Dictionary<string, string>
+            {
+                { "client_api", "NServiceBus" },
+                { "nservicebus_version", nsbFileVersion },
+                { "nservicebus.rabbitmq_version", rabbitMQFileVersion },
+                { "application", applicationName },
+                { "application_location", applicationPath },
+                { "machine_name", hostname },
+                { "user", userName },
+                { "endpoint_name", endpointName },
+            };
+
+            if (invalidOptionsMessage.Length > 0)
+            {
+                var message = invalidOptionsMessage.ToString().TrimEnd('\r', '\n');
+
+                Logger.Error(message);
+
+                throw new NotSupportedException(message);
+            }
+
+            return new ConnectionConfiguration(
+                host, port, virtualHost, userName, password, requestedHeartbeat, retryDelay, useTls, certPath, certPassPhrase, clientProperties);
+        }
+
+        static string GetValue(IDictionary<string, string> dictionary, string key, string @default)
```

Comment:
Could we change the name of the default parameter to `defaultValue` to get rid of needing the `@` syntax?